nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/SConv2D.java

```diff
@@ -40,20 +40,30 @@
 public class SConv2D extends Conv2D {
 
     @Builder(builderMethodName = "sameDiffSBuilder")
-    public SConv2D(SameDiff sameDiff, SDVariable[] inputFunctions, Conv2DConfig conv2DConfig) {
+    public SConv2D(SameDiff sameDiff, SDVariable[] inputFunctions,@NonNull Conv2DConfig conv2DConfig) {
         super(sameDiff, inputFunctions, conv2DConfig);
     }
 
-    public SConv2D(INDArray[] inputs, INDArray[] outputs, Conv2DConfig config){
+    public SConv2D(@NonNull INDArray[] inputs, INDArray[] outputs,@NonNull Conv2DConfig config){
         super(inputs, outputs, config);
     }
 
-    public SConv2D(@NonNull INDArray input, @NonNull INDArray depthWeights, INDArray pointWeights, INDArray bias, INDArray output, @NonNull Conv2DConfig config){
-        this(wrapFilterNull(input, depthWeights, pointWeights, bias), wrapOrNull(output), config);
+    public SConv2D(@NonNull INDArray layerInput, @NonNull INDArray depthWeights,@NonNull INDArray pointWeights,@NonNull INDArray bias, @NonNull Conv2DConfig Conv2DConfig){
+        this(wrapFilterNull(layerInput, depthWeights, pointWeights, bias), null, Conv2DConfig);
+    }
+
+    public SConv2D(@NonNull INDArray layerInput, @NonNull INDArray depthWeights,@NonNull INDArray pointWeights, @NonNull Conv2DConfig Conv2DConfig){
+        this(wrapFilterNull(layerInput, depthWeights, pointWeights), null, Conv2DConfig);
     }
 
     public SConv2D() {}
 
+    public SConv2D(@NonNull INDArray inputs,@NonNull Conv2DConfig conv2DConfig) {
+        super(inputs, conv2DConfig);
+    }
```

Comment:
Delete this constructor - depthWeights are always required. Bias and point weights are not.
You can see this from the c++ op definition:
https://github.com/eclipse/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/convo/sconv2d.cpp#L33-L51